### DISCUSSION TITLE
Multiple zenohc.hxx includes across project results in multiple definition of `zenohc::Publisher::get_keyexpr() const'

### DISCUSSION THREAD

**Discussion by lucasw** [eclipse-zenoh/roadmap#105] [Q&A]
The changes are in https://github.com/lucasw/ros_comm/commit/9b49c04a8cf1d02bb44bd4e78fb0c6a90682d784

I tried the include in ros.h, and that works, and it works in roscpp cpp files but not topic_manager.h, publisher.h, or topic.h (haven't tried others yet)- did I build zenoh cpp incorrectly?

```
Errors     << roscpp:make /home/lucasw/ros/ros1_zenoh/logs/roscpp/build.make.092.log                                                                                                                                                    
/usr/bin/ld: CMakeFiles/roscpp.dir/src/libros/publisher_link.cpp.o: in function `zenohc::Publisher::get_keyexpr() const':
publisher_link.cpp:(.text+0xf30): multiple definition of `zenohc::Publisher::get_keyexpr() const'; CMakeFiles/roscpp.dir/src/libros/subscriber.cpp.o:subscriber.cpp:(.text+0x780): first defined here
/usr/bin/ld: CMakeFiles/roscpp.dir/src/libros/publisher_link.cpp.o: in function `zenohc::Subscriber::get_keyexpr() const':
publisher_link.cpp:(.text+0xfa0): multiple definition of `zenohc::Subscriber::get_keyexpr() const'; CMakeFiles/roscpp.dir/src/libros/subscriber.cpp.o:subscriber.cpp:(.text+0x7f0): first defined here
/usr/bin/ld: CMakeFiles/roscpp.dir/src/libros/service_server.cpp.o: in function `zenohc::Publisher::get_keyexpr() const':
```

**Comment by lucasw** [comment]
Here's a much simpler example (than ros_comm) of the problem: https://github.com/lucasw/ros_one2z/tree/zenoh_cpp_ros_comm/ros1_example_pkg, where I added `#include <zenohc.hxx>` to a second file and get the multiple definition:

[zenoh_generate_image.hpp#L15](https://github.com/lucasw/ros_one2z/blob/3f29cc4aba7111048911b51536854637d93d6aeb/ros1_example_pkg/include/ros1_example_pkg/zenoh_generate_image.hpp#L15) and [bouncing_ball.cpp#L13](https://github.com/lucasw/ros_one2z/blob/3f29cc4aba7111048911b51536854637d93d6aeb/ros1_example_pkg/src/bouncing_ball.cpp#L13)

**Comment by Mallets** [comment]
Putting in the loop @milyin and @p-avital

**Reply by lucasw** [reply]
This change got it to link: https://github.com/eclipse-zenoh/zenoh-cpp/pull/87
